elastalert/kibana_discover.py

```diff
@@ -0,0 +1,192 @@
+            'use_kibana_discover was configured without kibana_discover_version for rule %s' % (
+                rule.get('name', '<MISSING NAME>')
+            )
+        )
+        return None
+
+    discover_url = rule.get('kibana_discover_url')
+    if not discover_url:
+        logging.warning(
+            'use_kibana_discover was configured without kibana_discover_url for rule %s' % (
+                rule.get('name', '<MISSING NAME>')
+            )
+        )
+        return None
+
+    index = rule.get('kibana_discover_index_pattern_id')
+    if not index:
+        logging.warning(
+            'use_kibana_discover was configured without kibana_discover_index_pattern_id for rule %s' % (
+                rule.get('name', '<MISSING NAME>')
+            )
+        )
+        return None
+
+    columns = rule.get('kibana_discover_columns', ['_source'])
+    filters = rule.get('filter', [])
+
+    if 'query_key' in rule:
+        query_keys = rule.get('compound_query_key', [rule['query_key']])
+    else:
+        query_keys = []
+
+    timestamp = lookup_es_key(match, rule['timestamp_field'])
+    timeframe = rule.get('timeframe', kibana_default_timedelta)
+    from_timedelta = rule.get('kibana_discover_from_timedelta', timeframe)
+    from_time = ts_add(timestamp, -from_timedelta)
+    to_timedelta = rule.get('kibana_discover_to_timedelta', timeframe)
+    to_time = ts_add(timestamp, to_timedelta)
+
+    if kibana_version in kibana5_kibana6_versions:
+        globalState = kibana6_disover_global_state(from_time, to_time)
+        appState = kibana_discover_app_state(index, columns, filters, query_keys, match)
+
+    elif kibana_version in kibana7_versions:
+        globalState = kibana7_disover_global_state(from_time, to_time)
+        appState = kibana_discover_app_state(index, columns, filters, query_keys, match)
+
+    else:
+        logging.warning(
+            'Unknown kibana discover application version %s for rule %s' % (
+                kibana_version,
+                rule.get('name', '<MISSING NAME>')
+            )
+        )
+        return None
+
+    return "%s?_g=%s&_a=%s" % (
+        os.path.expandvars(discover_url),
+        urllib.parse.quote(globalState),
+        urllib.parse.quote(appState)
+    )
+
+
+def kibana6_disover_global_state(from_time, to_time):
+    return prison.dumps( {
+        'refreshInterval': {
+            'pause': True,
+            'value': 0
+        },
+        'time': {
+            'from': from_time,
+            'mode': 'absolute',
+            'to': to_time
+        }
+    } )
+
+
+def kibana7_disover_global_state(from_time, to_time):
+    return prison.dumps( {
+        'filters': [],
+        'refreshInterval': {
+            'pause': True,
+            'value': 0
+        },
+        'time': {
+            'from': from_time,
+            'to': to_time
+        }
+    } )
+
+
+def kibana_discover_app_state(index, columns, filters, query_keys, match):
+    app_filters = []
+
```

Comment:
For simplicity, I just wrap up all the filters in a `bool` query.

![image](https://user-images.githubusercontent.com/12101120/65260019-b029e100-dad3-11e9-9829-33ba77357ec4.png)

![image](https://user-images.githubusercontent.com/12101120/65257836-e1a0ad80-dacf-11e9-9a85-054e611551bb.png)

Using Kibana's Sample web logs:

```
name: Errors
index: kibana_sample_data_logs

timestamp_field: "timestamp"
timestamp_type: "custom"
timestamp_format: "%Y-%m-%dT%H:%M:%S.%fZ"

type: any

filter:
- match:
    tags:
        query: error

use_kibana_discover: true
kibana_discover_url: http://localhost:5601/#/discover
kibana_discover_version: '6.8'
kibana_discover_index_pattern_id: 90943e30-9a47-11e8-b64d-95841ca0b247 # specific to my instance
kibana6_discover_columns: [ 'message', 'request' ]

alert: [ debug ]
alert_text: '{0}'
alert_text_args: [ kibana_link ]
alert_text_type: alert_text_only
```